src/understand-tidb/privilege-management.md

```diff
@@ -1,2 +1,65 @@
+
+## Adding privilege checks to a statement
+
+Some privilege checks are automatically assigned during plan building, for example ensuring that you have permissions to the tables that will be accessed. These checks are skipped for `information_schema` tables, and should you add an additional statement (such as `SHOW xyz`), you will also need to ensure that privilege checks are added.
+
+Should you need to add privilege checks there are two options:
+
+1. During plan building you can attach `visitInfo` to the plan (examples: [`SET CONFIG`](https://github.com/pingcap/tidb/blob/5e05922de6a253859cfbfe19356de8a2e2db39da/planner/core/planbuilder.go#L745), [`SHOW BACKUPS`](https://github.com/pingcap/tidb/blob/5e05922de6a253859cfbfe19356de8a2e2db39da/planner/core/planbuilder.go#L2378-L2380))
+
+2. In the executor function which handles the statement (examples: [`SHOW PROCESSLIST`](https://github.com/pingcap/tidb/blob/1a54708a7f8f86515236626c78e97a33d8adf583/executor/show.go#L368-L380)).
+
+The first option is recommended, as it is much less verbose. However, `visitInfo` does not handle cases where the statement can behave differently depending on the permissions of the user executing it. All users can execute the `SHOW PROCESSLIST` statement, but to see the sessions of other users requires the `PROCESS` privilege.
+
+`visitInfo` also does not support the case that a statement requires either `XYZ` **or** `ZYX` as a permission. An example of this, is that dropping a user in MySQL requires either the `CREATE USER` privilege, or the `DELETE` privilege on the `mysql.user` table. Permissions can only be attached in `visitInfo` with an **AND** semantic.
+
+### Manually checking with the privilege manager
+
+For (2) above, manual checks should follow the following pattern:
+
+```go
+checker := privilege.GetPrivilegeManager(e.ctx)
+if checker != nil && !checker.RequestVerification(ctx.GetSessionVars().ActiveRoles, schema.Name.L, table.Name.L, "", mysql.AllPrivMask) {
+    /* do something */
+}
+..
+if checker == nil || !checker.RequestDynamicVerification(ctx.GetSessionVars().ActiveRoles, "RESTRICTED_TABLES_ADMIN", false) {
+    /* do something */
+}
+```
+
+The check for `checker != nil` is important because for internal SQL statements the privilege manager is not present. These statements are expected to fall through and satisfy the privilege check.
+
+### Static and dynamic privileges
+
+Privileges fall into two categories:
+* Static privileges: These are the "traditional" privileges such as `INSERT`, `UPDATE`, `SELECT`, `DELETE`, `SUPER`, `PROCESS` which have existed in MySQL for a long time. They can *usually* be assigned to a user on either a global or database/table level.
```

Comment:
Please add a blank line before the lists.

Reply:
It's better to keep the same style for the lists, so please replace "`*`" with "`-`".

Reply:
I think most list among our guide using `*`